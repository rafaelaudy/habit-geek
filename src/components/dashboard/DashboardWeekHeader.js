import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faChevronCircleLeft,
  faChevronCircleRight
} from "@fortawesome/free-solid-svg-icons";
import { getWeekIntervalText } from "@rafael.audy/habit-geek-utils/utils/dateUtils";

const DashboardWeekHeader = ({
  week,
  isCurrentWeek,
  hasPreviousWeek,
  toggleWeek
}) => {
  const clickHandler = () => toggleWeek(!isCurrentWeek);
  const getButton = (direction, icon, isDisabled) => (
    <button
      className={`icon-button icon-button-${direction} btn btn-link ${
        isDisabled ? "disabled" : ""
      }`}
      onClick={clickHandler}
    >
      <FontAwesomeIcon icon={icon} />
    </button>
  );

  return (
    <div className="mb-3 d-flex justify-content-center align-items-baseline">
      {hasPreviousWeek
        ? getButton("previous", faChevronCircleLeft, !isCurrentWeek)
        : null}
      <h5 className="dashboard-week-header">{getWeekIntervalText(week)}</h5>
      {hasPreviousWeek
        ? getButton("forward", faChevronCircleRight, isCurrentWeek)
        : null}
    </div>
  );
};

export default DashboardWeekHeader;
