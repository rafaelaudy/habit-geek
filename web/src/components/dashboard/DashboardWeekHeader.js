import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faChevronCircleLeft,
  faChevronCircleRight
} from "@fortawesome/free-solid-svg-icons";
import { getWeekIntervalText } from "@habit-geek/shared/utils/dateUtils";

const DashboardWeekHeader = ({
  week,
  isCurrentWeek,
  hasPreviousWeek,
  toggleWeek
}) => {
  const clickHandler = () => toggleWeek(!isCurrentWeek);
  const getButton = (direction, icon, isDisabled) => (
    <button
      className={`habit__cell-action-container habit__cell-action-${direction} btn btn-link ${
        isDisabled ? "disabled" : ""
      }`}
      onClick={clickHandler}
    >
      <FontAwesomeIcon icon={icon} />
    </button>
  );

  return (
    <div className="mb-3 d-flex justify-content-center align-items-baseline">
      {hasPreviousWeek ? getButton('previous', faChevronCircleLeft, !isCurrentWeek) : null}
      <h5 className="dashboard-week-header__header">
        {getWeekIntervalText(week)}
      </h5>
      {hasPreviousWeek ? getButton('forward', faChevronCircleRight, isCurrentWeek) : null}
    </div>
  );
};

export default DashboardWeekHeader;
