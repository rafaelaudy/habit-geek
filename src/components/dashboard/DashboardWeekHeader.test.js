import React from "react";
import { shallow } from "enzyme";
import DashboardWeekHeader from "./DashboardWeekHeader";

const props = {
  week: "y1w1",
  isCurrentWeek: true,
  hasPreviousWeek: true,
  toggleWeek: () => {}
};

jest.mock("@rafael.audy/habit-geek-utils/utils/dateUtils.js", () => ({
  getWeekIntervalText: jest.fn().mockReturnValue("week 1: 30/Sep ... 6/Oct"),
}));

describe("DashboardWeekHeader component", () => {
  it("Renders static elements", () => {
    const dashboardWeekHeader = shallow(<DashboardWeekHeader {...props} />);
    expect(dashboardWeekHeader).toMatchSnapshot();
  });

  it("doesn't show buttons to change week when there are no previous weeks", () => {
    const dashboardWeekHeader = shallow(
      <DashboardWeekHeader {...props} hasPreviousWeek={false} />
    );
    expect(dashboardWeekHeader).toMatchSnapshot();
  });

  it("toggles the selected week", () => {
    const toggleWeekMock = jest.fn();
    const dashboardWeekHeader = shallow(
      <DashboardWeekHeader
        {...props}
        isCurrentWeek={false}
        toggleWeek={toggleWeekMock}
      />
    );
    dashboardWeekHeader
      .find("button")
      .at(0)
      .simulate("click");
    expect(toggleWeekMock).toHaveBeenCalledWith(true);
    expect(dashboardWeekHeader).toMatchSnapshot();
  });
});
