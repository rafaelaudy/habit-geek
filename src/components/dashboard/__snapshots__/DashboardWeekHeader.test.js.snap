// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardWeekHeader component Renders static elements 1`] = `
<div
  className="mb-3 d-flex justify-content-center align-items-baseline"
>
  <button
    className="icon-button icon-button-previous btn btn-link "
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f137",
            "M256 504C119 504 8 393 8 256S119 8 256 8s248 111 248 248-111 248-248 248zM142.1 273l135.5 135.5c9.4 9.4 24.6 9.4 33.9 0l17-17c9.4-9.4 9.4-24.6 0-33.9L226.9 256l101.6-101.6c9.4-9.4 9.4-24.6 0-33.9l-17-17c-9.4-9.4-24.6-9.4-33.9 0L142.1 239c-9.4 9.4-9.4 24.6 0 34z",
          ],
          "iconName": "chevron-circle-left",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </button>
  <h5
    className="dashboard-week-header"
  >
    week 1: 30/Sep ... 6/Oct
  </h5>
  <button
    className="icon-button icon-button-forward btn btn-link disabled"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f138",
            "M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm113.9 231L234.4 103.5c-9.4-9.4-24.6-9.4-33.9 0l-17 17c-9.4 9.4-9.4 24.6 0 33.9L285.1 256 183.5 357.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L369.9 273c9.4-9.4 9.4-24.6 0-34z",
          ],
          "iconName": "chevron-circle-right",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </button>
</div>
`;

exports[`DashboardWeekHeader component doesn't show buttons to change week when there are no previous weeks 1`] = `
<div
  className="mb-3 d-flex justify-content-center align-items-baseline"
>
  <h5
    className="dashboard-week-header"
  >
    week 1: 30/Sep ... 6/Oct
  </h5>
</div>
`;

exports[`DashboardWeekHeader component toggles the selected week 1`] = `
<div
  className="mb-3 d-flex justify-content-center align-items-baseline"
>
  <button
    className="icon-button icon-button-previous btn btn-link disabled"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f137",
            "M256 504C119 504 8 393 8 256S119 8 256 8s248 111 248 248-111 248-248 248zM142.1 273l135.5 135.5c9.4 9.4 24.6 9.4 33.9 0l17-17c9.4-9.4 9.4-24.6 0-33.9L226.9 256l101.6-101.6c9.4-9.4 9.4-24.6 0-33.9l-17-17c-9.4-9.4-24.6-9.4-33.9 0L142.1 239c-9.4 9.4-9.4 24.6 0 34z",
          ],
          "iconName": "chevron-circle-left",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </button>
  <h5
    className="dashboard-week-header"
  >
    week 1: 30/Sep ... 6/Oct
  </h5>
  <button
    className="icon-button icon-button-forward btn btn-link "
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f138",
            "M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm113.9 231L234.4 103.5c-9.4-9.4-24.6-9.4-33.9 0l-17 17c-9.4 9.4-9.4 24.6 0 33.9L285.1 256 183.5 357.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L369.9 273c9.4-9.4 9.4-24.6 0-34z",
          ],
          "iconName": "chevron-circle-right",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </button>
</div>
`;
